x/allocation/keeper/validator.go

```diff
@@ -0,0 +1,32 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/noislabs/noisd/x/allocation/types"
+)
+
+func (k Keeper) SetValidatorRewards(ctx sdk.Context, operator sdk.AccAddress, rewards types.ValidatorReward) {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, types.ValidatorRewardsPrefix)
+	bz := k.cdc.MustMarshal(&rewards)
+	prefixStore.Set(operator, bz)
+}
+
+func (k Keeper) DeleteValidatorRewards(ctx sdk.Context, operator sdk.AccAddress) {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, types.ValidatorRewardsPrefix)
+	prefixStore.Delete(operator)
+}
+
+func (k Keeper) GetValidatorRewards(ctx sdk.Context, operator sdk.AccAddress) types.ValidatorReward {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, types.ValidatorRewardsPrefix)
+	if !prefixStore.Has(operator) {
+		return types.ValidatorReward{}
+	}
+	bz := prefixStore.Get(operator)
```

Comment:
Do we need the extra `.Has` here? We should get `nil` or empty `bz` in that case and can check that.

Reply:
added nil check